src/frequenz/sdk/actor/power_distributing/power_distributing.py

```diff
@@ -62,6 +64,78 @@ def _get_all(source: dict[int, frozenset[int]], keys: abc.Set[int]) -> set[int]:
     return set().union(*[source[key] for key in keys])
 
 
+def _get_battery_inverter_mappings(
+    batteries: Iterable[int],
+    *,  # force keyword arguments
+    inv_bats: bool = True,
+    bat_bats: bool = True,
+    inv_invs: bool = True,
+) -> dict[str, dict[int, frozenset[int]]]:
+    """Create maps between battery and adjacent inverters.
+
+    Args:
+        batteries: batteries to create maps for
+        inv_bats: whether to create inverter to battery map
+        bat_bats: whether to create battery to battery map
+        inv_invs: whether to create inverter to inverter map
+
+    Returns:
+        a dict of the requested maps, using the following keys:
+            * "bat_invs": battery to inverters map
+            * "inv_bats": inverter to batteries map
+            * "bat_bats": battery to batteries map
+            * "inv_invs": inverter to inverters map
+    """
+    bat_invs_map: dict[int, set[int]] = {}
+    inv_bats_map: dict[int, set[int]] | None = {} if inv_bats else None
+    bat_bats_map: dict[int, set[int]] | None = {} if bat_bats else None
+    inv_invs_map: dict[int, set[int]] | None = {} if inv_invs else None
+    component_graph = connection_manager.get().component_graph
+
+    for battery_id in batteries:
+        inverters: set[int] = set(
+            component.component_id
+            for component in component_graph.predecessors(battery_id)
+            if component.category == ComponentCategory.INVERTER
+        )
+
+        if len(inverters) == 0:
+            _logger.error("No inverters for battery %d", battery_id)
+            continue
+
+        _logger.debug("Battery %d has inverter %s", battery_id, list(inverters))
```

Comment:
If there could be more than one inverter
```suggestion
        _logger.debug("Battery %d has inverter(s) %s", battery_id, list(inverters))
```